### PR TITLE
fix: add notification policy

### PR DESCRIPTION
**Pull request by stebenz**

### CODE REVIEW

File: examples/provider/resources/notification_policy.tf [hunk 1]
```diff
@@ -0,0 +1,6 @@
+resource zitadel_notification_policy notification_policy {
+  depends_on = [zitadel_org.org]
```
Comment (eliobischof) [inline examples/provider/resources/notification_policy.tf:2]: depends_on is not needed zitadel_org.org.id is referenced below

**Review by eliobischof — CHANGES_REQUESTED** [review]

**Review by eliobischof — CHANGES_REQUESTED** [review]
I think docs need to be generated?

File: zitadel/v2/default_notification_policy/funcs.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	if !ok {
+		return diag.Errorf("failed to get client")
+	}
+
+	client, err := helper.GetAdminClient(clientinfo)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	resp, err := client.UpdateNotificationPolicy(ctx, &admin.UpdateNotificationPolicyRequest{
```
[the remark this answers is not in the document]
Comment (stebenz) [reply zitadel/v2/default_notification_policy/funcs.go:32]: will add it, but is not necessary as there is no other possibility as when the passwordChange is changed with it as the only content attribute

File: zitadel/v2/notification_policy/funcs.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+		return diag.Errorf("failed to get client")
+	}
+
+	client, err := helper.GetManagementClient(clientinfo, d.Get(orgIDVar).(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	org := d.Get(orgIDVar).(string)
+
+	_, err = client.UpdateCustomNotificationPolicy(ctx, &management.UpdateCustomNotificationPolicyRequest{
```
[the remark this answers is not in the document]
Comment (stebenz) [reply zitadel/v2/notification_policy/funcs.go:48]: will add it, but is not necessary as there is no other possibility as when the passwordChange is changed with it as the only content attribute

File: examples/provider/resources/notification_policy.tf (hunk 1 above)
Comment (stebenz) [reply examples/provider/resources/notification_policy.tf:2]: will also remove it on all other examples, as the dependency is defined with the use of attributes of other resources

**Review by eliobischof — APPROVED** [review]

**Pull request merged**